rest_api/simple_supply_rest_api/transaction_creation.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2018 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# -----------------------------------------------------------------------------
+
+import hashlib
+
+from sawtooth_rest_api.protobuf import batch_pb2
+from sawtooth_rest_api.protobuf import transaction_pb2
+
+from simple_supply_addressing import addresser
+
+from simple_supply_protobuf import payload_pb2
+
+
+def make_create_agent_transaction(transaction_signer,
+                                  batch_signer,
+                                  name,
+                                  timestamp):
+    """Make a CreateAgentAction transaction and wrap it in a batch
+
+    Args:
+        transaction_signer (sawtooth_signing.Signer): The transaction key pair
+        batch_signer (sawtooth_signing.Signer): The batch key pair
+        name (str): The agent's name
+        timestamp (int): Unix UTC timestamp of when the agent is created
+
+    Returns:
+        batch_pb2.Batch: The transaction wrapped in a batch
+
+    """
+
+    inputs = [addresser.get_agent_address(
```

Comment:
Is this supposed to be the same list? You could just say `address_list = [addresser...]` and then later do `inputs=address_list` and `outputs=address_list`.

Reply:
This is true, but not in the general case for Simple Supply transactions. My reasoning for doing it this way was to make it as general and easy to follow as possible. It may not be obvious to someone that the inputs and outputs for a transaction are not always the same. But I could see an argument for changing it. Let me know what you think.